elliottlib/bzutil.py

```diff
@@ -23,31 +27,118 @@
 logger = logutil.getLogger(__name__)
 
 
-class BugzillaBugTracker:
-    def login(self, interactive_login):
-        client = bugzilla.Bugzilla(self._server)
-        if not client.logged_in:
-            print(f"elliott requires cached login credentials for {self._server}")
-            if interactive_login:
-                self._client.interactive_login()
-            else:
-                red_print("Login using 'bugzilla login --api-key'")
-                exit(1)
-        return client
+class Bug(ABC):
+    def __init__(self, bug_obj):
+        self.bug = bug_obj
+
+
+class BugzillaBug(Bug):
+    def __getattr__(self, attr):
+        if attr in self.__dict__:
+            return getattr(self, attr)
+        return getattr(self.bug, attr)
+
+    def __init__(self, bug_obj):
+        super().__init__(bug_obj)
+        self.creation_time_parsed = datetime.strptime(str(self.bug.creation_time), '%Y%m%dT%H:%M:%S')
 
-    def __init__(self, config, interactive_login=False):
+
+class JIRABug(Bug):
+    def _url(self):
+        o = urlparse(self.bug.self)
+        return o._replace(path=f"/browse/{self.bug_id}").geturl()
+
+    def __init__(self, bug_obj):
+        super().__init__(bug_obj)
+        self.bug_id = self.bug.key
+        self.weburl = self._url()
+        self.component = self.bug.fields.components[0].name
+        self.status = self.bug.fields.status.name
+        self.creation_time_parsed = datetime.strptime(str(self.bug.fields.created), '%Y-%m-%dT%H:%M:%S.%f%z')
+        self.summary = self.bug.fields.summary
+
+
+class BugTracker(ABC):
+    def __init__(self, config):
         self.config = config
         self._server = config.get('server')
-        self._client = self.login(interactive_login)
 
     def target_release(self):
         return self.config.get('target_release')
 
+    @abstractmethod
+    def search(self):
+        raise NotImplementedError
+
+
+class JIRABugTracker(BugTracker):
+    @staticmethod
+    def get_config(runtime):
+        return runtime.gitdata.load_data(key='jira').data
+
+    def login(self, token_auth=None) -> JIRA:
+        if not token_auth:
+            token_auth = os.environ.get("JIRA_TOKEN")
```

Comment:
we should check/set proxy

Reply:
Let's make that a separate PR. But thanks for noting that